### PR TITLE
Improvements to temperature panel resizing

### LINKED ISSUE #177: Improve resizing of widgets on right-hand side of GUI
Currently, only the OPUS panel expands vertically, while the temperature plot and temperature monitoring/control panels are pushed downwards. This leaves the temperature plot relatively small, and the OPUS panel unnecessarily large. 

Either fix the size of the OPUS panel vertically, with the temperature plot free to expand, or let them both expand.

### PR DESCRIPTION
**Pull request by dc2917**
This PR includes some improvements to the resizing of panels on the right-hand side of the GUI, as well as an aesthetic improvement to the buttons which toggle plot visibility.

Upon resizing of the window, only the panels containing the OPUS view and temperature plot are now free to expand vertically. However, as the temperature plot panel contains a `QGridLayout`, which contains widgets with different `QSizePolicy`s, a custom resize function is required to maximise the space taken up by the figure.

Closes #177

### CODE REVIEW

**Review by alexdewar — CHANGES_REQUESTED** [review]
This looks much nicer. Good work! I think we should probs also move the OPUS panel somewhere else (maybe into the gap in the lefthand layout?) and then the plot can be even larger and then it'll look great. That's a job for another PR though.

I've made one small suggestion, but other than that I'm happy with it. The only thing is that I'm a bit concerned that the custom resize function feels like a bit of a hack. It's entirely possible that that's the only way to do what we want, but I've requested a review from @dalonsoa anyway to see if he's got any better ideas.

**Comment by dc2917** [comment]
I've just found an issue with the resizing, so this needs a bit more work before reviews.

**Comment by dc2917** [comment]
> This looks much nicer. Good work! I think we should probs also move the OPUS panel somewhere else (maybe into the gap in the lefthand layout?) and then the plot can be even larger and then it'll look great. That's a job for another PR though.
> 
> I've made one small suggestion, but other than that I'm happy with it. The only thing is that I'm a bit concerned that the custom resize function feels like a bit of a hack. It's entirely possible that that's the only way to do what we want, but I've requested a review from @dalonsoa anyway to see if he's got any better ideas.

Yeah I completely agree that it feels like a hack. I was happy that I could at least set the figure position and size based on those of other widgets, which are decided by Qt.

I think the issue is that I set a minimum size for the figure. If I don't do this, then that panel is tiny by default - Qt seems to set its height to be the vertical distance between the top of the "Hot BB" button and the bottom of the "Cold BB" button. This itself, I think, is due to the size policies of the layouts. But we've seen what happens if we let Qt choose the size policies - the OPUS panel gets all the space. For this same reason, if I don't set the size policy of either the "Hot BB" and "Cold BB" buttons or the figure, the buttons get all the space. 

In short, it seems we do need to override some Qt behaviour. I'm sure there will be improvements to the way I've done this, but I think something of this sort is necessary.

File: finesse/gui/main_window.py [hunk 1]
```diff
@@ -65,6 +72,18 @@ def __init__(self) -> None:
...
+            panel.setSizePolicy(
+                QSizePolicy.Policy.Preferred,
+                QSizePolicy.Policy.MinimumExpanding,
+            )
+        for panel in [dp9800, tc4820_hot, tc4820_cold, data_file]:
+            panel.setSizePolicy(
+                QSizePolicy.Policy.Preferred,
+                QSizePolicy.Policy.Fixed,
+            )
+
```
[the remark this answers is not in the document]
Comment (dc2917) [reply finesse/gui/main_window.py:86]: Even though we've made it so that each component can be run as a standalone GUI?

[the remark this answers is not in the document]
Comment (alexdewar) [reply finesse/gui/main_window.py:86]: Yeah. I think that was fine for testing, but I don't know that we really need that capability now that things are better integrated. There are so many moving parts now I suspect most of the components won't actually work standalone now anyway, especially seeing as we don't have tests for those bits.

[the remark this answers is not in the document]
Comment (dc2917) [reply finesse/gui/main_window.py:86]: Makes sense!

**Review by alexdewar — APPROVED** [review]
Aside from a trivial merge conflict, I think this is good to go!

**Comment by dalonsoa** [comment]
Starting to renew this. It looks good, but the process is not that robust. This is what I get after a couple of resizes (in MacOS):

image

I'm investigating PySide for the correct combination of options to make this resizing fluid. It should not be necessary to have this hack.

**Comment by dc2917** [comment]
> Starting to renew this. It looks good, but the process is not that robust. This is what I get after a couple of resizes (in MacOS):
> 
> image
> 
> I'm investigating PySide for the correct combination of options to make this resizing fluid. It should not be necessary to have this hack.

Interesting, that looks like it's ignored (or perhaps overwritten) the minimum size setting.

**Comment by alexdewar** [comment]
@dc2917 What's happening with this PR? Did we decide not to merge this one and merge @dalonsoa's fix instead? If so, would you mind opening a new PR for that one into `main` please?

**Comment by dc2917** [comment]
Sorry @alexdewar this completely slipped my mind. Diego's fix has been merged into this one, I'll remove the custom resize function itself and push now.

**Comment by alexdewar** [comment]
@dc2917 Does anything else need doing or is this good to merge?
